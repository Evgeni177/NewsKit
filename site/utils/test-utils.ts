import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
