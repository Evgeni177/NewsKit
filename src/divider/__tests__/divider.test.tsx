import {renderToFragmentWithTheme} from '../../test/test-utils';
import {Divider} from '..';

describe('Divider', () => {
  test('renders with default styles', () => {
    const fragment = renderToFragmentWithTheme(Divider);
    expect(fragment).toMatchSnapshot();
  });
  test('renders with props', () => {
    const fragment = renderToFragmentWithTheme(Divider, {
      $color: 'interface050',
      $border: 'borderWidth030',
      $marginTop: 'spacingSize040',
      $marginRight: 'spacingSize090',
      $marginBottom: 'spacingSize040',
      $marginLeft: 'spacingSize090',
    });

    expect(fragment).toMatchSnapshot();
  });
});
