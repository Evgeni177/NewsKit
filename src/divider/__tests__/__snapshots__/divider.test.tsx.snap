// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: #e4e4e4;
  border-width: 1px;
  margin-top: 24px;
  margin-right: 0;
  margin-bottom: 24px;
  margin-left: 0;
  border-style: solid;
}

<hr
    class="emotion-0"
    data-testid="divider"
  />
</DocumentFragment>
`;

exports[`Divider renders with props 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-color: #0a0a0a;
  border-width: 4px;
  margin-top: 16px;
  margin-right: 64px;
  margin-bottom: 16px;
  margin-left: 64px;
  border-style: solid;
}

<hr
    class="emotion-0"
    data-testid="divider"
  />
</DocumentFragment>
`;
